semgrep/semgrep/ignores.py

```diff
@@ -89,15 +89,15 @@ def _survives(self, path: Path) -> bool:
 
         return True
 
-    def filter_paths(self, *, candidates: Iterable[Path]) -> FilteredFiles:
-        kept, removed = partition(
-            candidates,
-            lambda path: path.exists()
-            and (
-                self._survives(path.absolute())
-                or path.absolute().samefile(self.base_path)
-            ),
+    @lru_cache(maxsize=100_000)
```

Comment:
Non-blocking question:
This maxsize seems effectively infinite, in which case, wouldn't we rather just use an in-memory data-structure?

Reply:
`@lru_cache` itself uses an in-memory data-structure; if you're suggesting implementing custom code for caching in a dict we maintain ourselves, I just prefer not doing that as this is less code. I could also set maxsize to `None` but I like setting a specific number to communicate "we thought about this and yes, we can cache all values that will ever show up"

Reply:
Right, the trap one gets into with a large size is that, when the access pattern has on the order of 100k new values before a cache hit, that the performance can drop once the list size crosses the threshold.

When I've worked with these cache patterns before, it's been more useful to, e.g., sort the input list, and use a much smaller cache size.

Reply:
Although in this case if someone is scanning 100,000 files surely they don't care about the couple hundred milliseconds of overhead added by caching-and-evicting compared to the actual time spent scanning? And especially compared to the slowdown caused by not having any caching.

I also can't think of a way of pre-sorting the entries with the biggest time loss.